src/dotnet/commands/dotnet-migrate/TemporaryDotnetNewTemplateProject.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Microsoft.Build.Construction;
+using Microsoft.DotNet.Cli;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Cli
+{
+    internal class TemporaryDotnetNewTemplateProject
+    {
+        private static string s_temporaryDotnetNewMSBuildProjectName = "p";
+
+        public TemporaryDotnetNewTemplateProject()
+        {
+            ProjectDirectory = CreateDotnetNewMSBuild(s_temporaryDotnetNewMSBuildProjectName);
+            MSBuildProject = GetMSBuildProject(ProjectDirectory);
+            ProjectJson = GetProjectJson(ProjectDirectory);
+        }
+
+        public ProjectRootElement MSBuildProject { get; }
+        public JObject ProjectJson { get; }
+        public string ProjectDirectory { get; }
+
+        public string ProjectJsonPath => Path.Combine(ProjectDirectory, "project.json");
+        public string MSBuildProjectPath => Path.Combine(ProjectDirectory, s_temporaryDotnetNewMSBuildProjectName);
+
+        public void Clean()
+        {
+            Directory.Delete(ProjectDirectory, true);
+        }
+
+        private string CreateDotnetNewMSBuild(string projectName)
+        {
+            var guid = Guid.NewGuid().ToString();
+            var tempDir = Path.Combine(
```

Comment:
`Path.Combine(Path.GetTempPath(), this.GetType().Namespace, Path.GetRandomFileName(), s_temporaryDotnetNewMSBuildProjectName)` ?

The `s_temporaryDotnetNewMSBuildProjectName` is needed to have predictable output file names. The namespace in the path makes debugging a little easier.